test/Unit/models/Import/Service/QtiResultXmlImporterTest.php

```diff
@@ -0,0 +1,243 @@
+            ->with($test)
+            ->willReturn($xmlDocument);
+
+        $delivery->expects($this->once())
+            ->method('getUri')
+            ->willReturn('deliveryId');
+
+        $deliveryExecution
+            ->expects($this->once())
+            ->method('getDelivery')
+            ->willReturn($delivery);
+
+        $test->expects($this->any())
+            ->method('getUri')
+            ->willReturn('testId');
+
+        $resource->expects($this->once())
+            ->method('getOnePropertyValue')
+            ->willReturn($test);
+
+        $item->expects($this->once())
+            ->method('getIdentifier')
+            ->willReturn('itemIdentifier');
+
+        $item->expects($this->once())
+            ->method('getHref')
+            ->willReturn('itemHref');
+
+        $assessmentTest->expects($this->once())
+            ->method('getComponentsByClassName')
+            ->with('assessmentItemRef')
+            ->willReturn($qtiComponentCollection);
+
+        $xmlDocument->expects($this->once())
+            ->method('getDocumentComponent')
+            ->willReturn($assessmentTest);
+
+        $resultStorage
+            ->expects($this->once())
+            ->method('storeTestVariables')
+            ->with(
+                'deliveryExecutionId',
+                'testId',
+                [$testOutcomeVariable],
+                'deliveryExecutionId'
+            );
+
+        $resultStorage
+            ->expects($this->once())
+            ->method('storeItemVariables')
+            ->with(
+                'deliveryExecutionId',
+                'testId',
+                'itemHref',
+                [$itemOutcomeVariable],
+                'deliveryExecutionId.itemIdentifier.0'
+            );
+
+        $this->sut->importQtiResultXml('deliveryExecutionId', $this->getXmlContent());
+    }
+
+    private function getResultMapperMock($testOutcomeVariable, $itemOutcomeVariable): ResultMapper
+    {
+        $resultMapperMock = $this->createMock(ResultMapper::class);
+
+        $resultMapperMock
+            ->expects($this->once())
+            ->method('getTestVariables')
+            ->willReturn(
+                [
+                    [$testOutcomeVariable],
+                ]
+            );
+
+        $resultMapperMock
+            ->expects($this->once())
+            ->method('getItemVariables')
+            ->willReturn(
+                [
+                    'itemIdentifier' => [$itemOutcomeVariable],
+                ]
+            );
+
+        return $resultMapperMock;
+    }
+
+    private function getXmlContent(): string
+    {
+        // phpcs:disable
+        return <<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<assessmentResult 
+    xmlns="http://www.imsglobal.org/xsd/imsqti_result_v2p1" 
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
+    xsi:schemaLocation="http://www.imsglobal.org/xsd/imsqti_result_v2p1 http://www.imsglobal.org/xsd/qti/qtiv2p1/imsqti_result_v2p1.xsd"
```

Comment:
Maybe splitting the schema URLs in different lines works here without needing to disable `phpcs`?

```suggestion
    xsi:schemaLocation="http://www.imsglobal.org/xsd/imsqti_result_v2p1
                                           http://www.imsglobal.org/xsd/qti/qtiv2p1/imsqti_result_v2p1.xsd"
```

Reply:
Thanks! In this case, since it is an XML code, not a PHP code, I would prefer to ignore the PHPCs validation for those cases, if you do not mind :)